tls/s2n_handshake_io.c

```diff
@@ -475,6 +471,21 @@ int s2n_generate_new_client_session_id(struct s2n_connection *conn)
     return 0;
 }
 
+/* Lets the server flag whether a HelloRetryRequest is needed while processing extensions */
+int s2n_set_hello_retry_required(struct s2n_connection *conn)
+{
+    conn->handshake.hello_retry_request = 1;
+
+    return 0;
+}
+
+/* Lets the server determine whether a HelloRetryRequest should be sent.
+ * A retry is only possible after the first ClientHello (HELLO_RETRY_MSG). */
+bool s2n_is_hello_retry_required(struct s2n_connection *conn)
+{
+    return (s2n_conn_get_current_message_type(conn) == HELLO_RETRY_MSG) && (conn->handshake.handshake_type & HELLO_RETRY_REQUEST);
```

Comment:
Why check the current message type as part of this? Is there any need to enforce what handler calls this method?

Reply:
I was trying to get something working with SAW. This isn't required for HRR to work though (removed the message check)